Src/AmrCore/AMReX_Interp_C.H

```diff
@@ -57,10 +61,14 @@ face_linear_face_interp_y (int fi, int fj, int fk, int n, Array4<T> const& fine,
 #else
         int ck = 0;
 #endif
-        // Check mask to ensure we don't overwrite valid fine data.
-        if (mask(ci, cj, ck, n)) {
-            fine(fi, fj, fk, n) = crse(ci, cj, ck, n);
+
+        if (mask) {
+            // Check solve mask to ensure we don't overwrite valid fine data.
+            if (!mask(ci, cj, ck, n))
+            { return; }
```

Comment:
I was first confused when reading this.  It might be more readable like this because it's shorter.
```
if (!mask || mask(ci,cj,ck,n)) {
    fine(...) = crse(...);
}
```

Reply:
I thought you once told me that you can't do that because some compilers will evaluate `!mask` and `mask(i,j,k,n)` at the same time and then have problems if mask is not allocated.

Reply:
thanks for clarifying